src/mlpack/core/cv/purged_k_fold_cv_impl.hpp

```diff
@@ -0,0 +1,376 @@
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+template<typename ElementType>
+arma::Mat<ElementType> PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::GetTrainingSubset(const arma::Mat<ElementType>& m, size_t i)
+{
+  return m.cols(GetTrainingSubsetCols(i));
+}
+
+template<
+    typename MLAlgorithm,
+    typename Metric,
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+template<typename ElementType>
+arma::Row<ElementType> PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::GetTrainingSubset(const arma::Row<ElementType>& r, size_t i)
+{
+  return r.cols(GetTrainingSubsetCols(i));
+}
+
+template<
+    typename MLAlgorithm,
+    typename Metric,
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+template<typename ElementType>
+arma::Mat<ElementType> PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::GetValidationSubset(const arma::Mat<ElementType>& m, size_t i)
+{
+  return m.cols(GetValidationSubsetCols(i));
+}
+
+template<
+    typename MLAlgorithm,
+    typename Metric,
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+template<typename ElementType>
+arma::Row<ElementType> PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::GetValidationSubset(const arma::Row<ElementType>& r, size_t i)
+{
+  return r.cols(GetValidationSubsetCols(i));
+}
+
+template<
+    typename MLAlgorithm,
+    typename Metric,
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+void PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::CheckIntervals() const
+{
+  if (intervals.n_rows != 2)
+    throw std::invalid_argument(
+        "PurgedKFoldCV::CheckIntervals(): "
+        "intervals must be a 2 x m matrix!");
+}
+
+template<
+    typename MLAlgorithm,
+    typename Metric,
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+arma::span PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::GetValidationSubsetCols(size_t i) const
+{
+  const size_t firstCol(ValidationSubsetFirstCol(i));
+
+  return arma::span(firstCol,
+      std::min(firstCol + this->binSize,
+          (this->k - 1) * this->binSize + this->lastBinSize) - 1);
+}
+
+template<
+    typename MLAlgorithm,
+    typename Metric,
+    typename MatType,
+    typename PredictionsType,
+    typename WeightsType>
+arma::uvec PurgedKFoldCV<
+    MLAlgorithm,
+    Metric,
+    MatType,
+    PredictionsType,
+    WeightsType>::GetTrainingSubsetCols(size_t i) const
+{
+  const arma::span  validationSubsetCols(GetValidationSubsetCols(i));
+  const arma::uword datasetSize(
+      (this->k - 1) * this->binSize + this->lastBinSize);

```

Comment:
Shouldn't this always be equal to `xs.n_cols`?  Couldn't you just use that instead here?

Reply:
Good spot